<?xml version="1.0"?>

<launch>
    <arg name="map_file" default="$(find chibi20_2)/mapdata/goodmap.yaml"/>
    <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)" output="screen"/>

    <!-- <include file="$(find chibi21_b)/launch/Localizer_path.launch" /> -->
    <param name="use_sim_time" value="true" type="bool"/>
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="&#45;d $(find distance_based_localizer)/config/simulate.rviz" /> -->
    <!-- 5 -->
    <!-- <node pkg="rosbag" type="play" name="play" args="&#45;&#45;clock /home/amsl/bagfiles/with_yolo2/yroomba1.bag /home/amsl/bagfiles/with_yolo2/rooomba2.bag /home/amsl/bagfiles/with_yolo2/rooomba3.bag /home/amsl/bagfiles/with_yolo2/rooomba4.bag /home/amsl/bagfiles/with_yolo2/rooomba5.bag" /> -->
    <!-- 4-->
    <!-- <node pkg="rosbag" type="play" name="play" args="&#45;&#45;clock /home/amsl/bagfiles/with_yolo2/yroomba1.bag /home/amsl/bagfiles/with_yolo2/rooomba2.bag /home/amsl/bagfiles/with_yolo2/rooomba3.bag /home/amsl/bagfiles/with_yolo2/rooomba4.bag" /> -->
    <!-- 3 -->
    <!-- <node pkg="rosbag" type="play" name="play" args="&#45;&#45;clock /home/amsl/bagfiles/with_yolo2/rooomba3.bag /home/amsl/bagfiles/with_yolo2/rooomba2.bag /home/amsl/bagfiles/with_yolo2/yroomba1.bag" /> -->
    <!-- 1 -->
    <!-- <node pkg="rosbag" type="play" name="play" args="&#45;&#45;clock /home/amsl/bagfiles/with_yolo2/rooomba3.bag" /> -->
    <!-- <node pkg="rosbag" type="play" name="play" args="&#45;&#45;clock /home/amsl/bagfiles/six_roomba_20210429/six_roomba_6_2021&#45;04&#45;29&#45;18&#45;56&#45;33.bag" /> -->
    <!-- 2 -->
    <!-- <node pkg="rosbag" type="play" name="play" args="&#45;&#45;clock /home/amsl/bagfiles/with_yolo2/yroomba1.bag /home/amsl/bagfiles/with_yolo2/rooomba2.bag" /> -->
    <!-- 6 -->
    <node pkg="rosbag" type="play" name="play" args="--clock /home/amsl/bagfiles/with_yolo2/yroomba1.bag /home/amsl/bagfiles/with_yolo2/rooomba2.bag /home/amsl/bagfiles/with_yolo2/rooomba3.bag /home/amsl/bagfiles/with_yolo2/rooomba4.bag /home/amsl/bagfiles/with_yolo2/rooomba5.bag /home/amsl/bagfiles/with_yolo2/rooomba6.bag" />

    <!-- roobma1 only odom -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_odom" name="distance_based_localizer_1_odom" >
        <remap from="/roomba/odometry" to="roomba1/roomba/odometry" />
        <remap from="/db_pose" to="roomba1_odom/db_pose" />
        <remap from="/db_poses" to="roomba1_odom/db_poses" />
        <remap from="/score" to="roomba1_odom/score" />
        <remap from="/path" to="roomba1_odom/path" />

        <param name="roomba_name" value="1" type="int" />
        <param name="num_s" value="1.0" type="double" />
        <param name="weight_s" value="1.5" type="double" />
        <param name="particlce_num" value="300" type="int" />
        <param name="x_init" value="7.0" type="double" />
        <param name="y_init" value="1.0" type="double" />
        <param name="yaw_init" value="1.57" type="double" />
        <param name="bench" value="1.0" type="double" />
        <param name="fire" value="1.0" type="double" />
        <param name="big" value="1.0" type="double" />
        <param name="trash" value="0.8" type="double" />
        <param name="odom" value="20" type="double" />
        <param name="alpha_slow_th" value="0.001" type="double" />
        <param name="alpha_fast_th" value="0.1" type="double" />
        <param name="x_cov_init" value="0.8" type="double" />
        <param name="y_cov_init" value="0.8" type="double" />
        <param name="yaw_cov_init" value="0.4" type="double" />
        <param name="x_cov_1" value="0.1" type="double" />
        <param name="y_cov_1" value="0.1" type="double" />
        <param name="yaw_cov_1" value="0.1" type="double" />
        <param name="x_cov_2" value="0.01" type="double" />
        <param name="y_cov_2" value="0.01" type="double" />
        <param name="yaw_cov_2" value="0.02" type="double" />
        <param name="limit" value="100" type="int" />
        <param name="move_noise" value="0.2" type="double" />
        <param name="estimated_weight_th" value="4.0" type="double" />
        <param name="roomba_odom" value="roomba1_odom_odom" type="string" />
        <param name="is_only_odom" value="true" type="bool" />
    </node>
    <!-- roomba2 only_odom -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_odom" name="distance_based_localizer_2_odom" output="screen" >
    <node pkg="distance_based_localizer" type="distance_based_localizer_odom" name="distance_based_localizer_2_odom" >
        <remap from="/roomba/odometry" to="roomba2/roomba/odometry" />
        <remap from="/db_pose" to="roomba2_odom/db_pose" />
        <remap from="/db_poses" to="roomba2_odom/db_poses" />
        <remap from="/score" to="roomba2_odom/score" />
        <remap from="/path" to="roomba2_odom/path" />

        <param name="roomba_name" value="2" type="int" />
        <param name="num_s" value="1.0" type="double" />
        <param name="weight_s" value="1.5" type="double" />
        <param name="particlce_num" value="300" type="int" />
        <param name="x_init" value="7.0" type="double" />
        <param name="y_init" value="0.5" type="double" />
        <param name="yaw_init" value="1.57" type="double" />
        <param name="bench" value="1.0" type="double" />
        <param name="fire" value="1.0" type="double" />
        <param name="big" value="1.0" type="double" />
        <param name="trash" value="0.8" type="double" />
        <param name="odom" value="20" type="double" />
        <param name="alpha_slow_th" value="0.001" type="double" />
        <param name="alpha_fast_th" value="0.1" type="double" />
        <param name="x_cov_init" value="0.8" type="double" />
        <param name="y_cov_init" value="0.8" type="double" />
        <param name="yaw_cov_init" value="0.4" type="double" />
        <param name="x_cov_1" value="0.1" type="double" />
        <param name="y_cov_1" value="0.1" type="double" />
        <param name="yaw_cov_1" value="0.1" type="double" />
        <param name="x_cov_2" value="0.01" type="double" />
        <param name="y_cov_2" value="0.01" type="double" />
        <param name="yaw_cov_2" value="0.02" type="double" />
        <param name="limit" value="100" type="int" />
        <param name="move_noise" value="0.2" type="double" />
        <param name="estimated_weight_th" value="4.0" type="double" />
        <param name="roomba_odom" value="roomba2_odom_odom" type="string" />
        <param name="is_only_odom" value="true" type="bool" />
        </node>
    <!-- roomba3_odom  -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_odom" name="distance_based_localizer_3_odom" >
        <remap from="/roomba/odometry" to="roomba3/roomba/odometry" />
        <remap from="/db_pose" to="roomba3_odom/db_pose" />
        <remap from="/db_poses" to="roomba3_odom/db_poses" />
        <remap from="/score" to="roomba3_odom/score" />
        <remap from="/path" to="roomba3_odom/path" />

        <param name="roomba_name" value="3" type="int" />
        <param name="num_s" value="1.0" type="double" />
        <param name="weight_s" value="1.5" type="double" />
        <param name="particlce_num" value="300" type="int" />
        <param name="x_init" value="7.0" type="double" />
        <param name="y_init" value="0.0" type="double" />
        <param name="yaw_init" value="1.57" type="double" />
        <param name="bench" value="1.0" type="double" />
        <param name="fire" value="1.0" type="double" />
        <param name="big" value="1.0" type="double" />
        <param name="trash" value="0.8" type="double" />
        <param name="odom" value="20" type="double" />
        <param name="alpha_slow_th" value="0.001" type="double" />
        <param name="alpha_fast_th" value="0.1" type="double" />
        <param name="x_cov_init" value="0.8" type="double" />
        <param name="y_cov_init" value="0.8" type="double" />
        <param name="yaw_cov_init" value="0.4" type="double" />
        <param name="x_cov_1" value="0.1" type="double" />
        <param name="y_cov_1" value="0.1" type="double" />
        <param name="yaw_cov_1" value="0.1" type="double" />
        <param name="x_cov_2" value="0.01" type="double" />
        <param name="y_cov_2" value="0.01" type="double" />
        <param name="yaw_cov_2" value="0.02" type="double" />
        <param name="limit" value="100" type="int" />
        <param name="move_noise" value="0.2" type="double" />
        <param name="estimated_weight_th" value="4.0" type="double" />
        <param name="roomba_odom" value="roomba3_odom_odom" type="string" />
        <param name="is_only_odom" value="true" type="bool" />
    </node>
    <!-- roomba4 only_odom -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_odom" name="distance_based_localizer_4_odom" output="screen" >
        <remap from="/roomba/odometry" to="roomba4/roomba/odometry" />
        <remap from="/db_pose" to="roomba4_odom/db_pose" />
        <remap from="/db_poses" to="roomba4_odom/db_poses" />
        <remap from="/score" to="roomba4_odom/score" />
        <remap from="/path" to="roomba4_odom/path" />

        <param name="roomba_name" value="4" type="int" />
        <param name="num_s" value="1.0" type="double" />
        <param name="weight_s" value="1.5" type="double" />
        <param name="particlce_num" value="300" type="int" />
        <param name="x_init" value="7.0" type="double" />
        <param name="y_init" value="-0.5" type="double" />
        <param name="yaw_init" value="1.57" type="double" />
        <param name="bench" value="1.0" type="double" />
        <param name="fire" value="1.0" type="double" />
        <param name="big" value="1.0" type="double" />
        <param name="trash" value="1.0" type="double" />
        <param name="odom" value="20" type="double" />
        <param name="alpha_slow_th" value="0.001" type="double" />
        <param name="alpha_fast_th" value="0.1" type="double" />
        <param name="x_cov_init" value="0.8" type="double" />
        <param name="y_cov_init" value="0.8" type="double" />
        <param name="yaw_cov_init" value="0.4" type="double" />
        <param name="x_cov_1" value="0.1" type="double" />
        <param name="y_cov_1" value="0.1" type="double" />
        <param name="yaw_cov_1" value="0.1" type="double" />
        <param name="x_cov_2" value="0.01" type="double" />
        <param name="y_cov_2" value="0.01" type="double" />
        <param name="yaw_cov_2" value="0.02" type="double" />
        <param name="limit" value="100" type="int" />
        <param name="move_noise" value="0.2" type="double" />
        <param name="estimated_weight_th" value="4.0" type="double" />
        <param name="roomba_odom" value="roomba4_odom_odom" type="string" />
        <param name="is_only_odom" value="true" type="bool" />
    </node>
    <!-- roomba5 only_odom -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_odom" name="distance_based_localizer_5_odom" output="screen" >
        <remap from="/roomba/odometry" to="roomba5/roomba/odometry" />
        <remap from="/db_pose" to="roomba5_odom/db_pose" />
        <remap from="/db_poses" to="roomba5_odom/db_poses" />
        <remap from="/score" to="roomba5_odom/score" />
        <remap from="/path" to="roomba5_odom/path" />

        <param name="roomba_name" value="5" type="int" />
        <param name="num_s" value="1.0" type="double" />
        <param name="weight_s" value="1.5" type="double" />
        <param name="particlce_num" value="300" type="int" />
        <param name="x_init" value="7.0" type="double" />
        <param name="y_init" value="-1.0" type="double" />
        <param name="yaw_init" value="1.57" type="double" />
        <param name="bench" value="1.0" type="double" />
        <param name="fire" value="1.0" type="double" />
        <param name="big" value="1.0" type="double" />
        <param name="trash" value="0.8" type="double" />
        <param name="odom" value="20" type="double" />
        <param name="alpha_slow_th" value="0.001" type="double" />
        <param name="alpha_fast_th" value="0.1" type="double" />
        <param name="x_cov_init" value="0.8" type="double" />
        <param name="y_cov_init" value="0.8" type="double" />
        <param name="yaw_cov_init" value="0.4" type="double" />
        <param name="x_cov_1" value="0.1" type="double" />
        <param name="y_cov_1" value="0.1" type="double" />
        <param name="yaw_cov_1" value="0.1" type="double" />
        <param name="x_cov_2" value="0.01" type="double" />
        <param name="y_cov_2" value="0.01" type="double" />
        <param name="yaw_cov_2" value="0.02" type="double" />
        <param name="limit" value="100" type="int" />
        <param name="move_noise" value="0.2" type="double" />
        <param name="estimated_weight_th" value="4.0" type="double" />
        <param name="roomba_odom" value="roomba5_odom_odom" type="string" />
        <param name="is_only_odom" value="true" type="bool" />
    </node>
    <!--  -->
<!-- roomba6 only_odom  -->
    <node pkg="distance_based_localizer" type="distance_based_localizer_odom" name="distance_based_localizer_6_odom" output="screen" >
        <remap from="/roomba/odometry" to="roomba6/roomba/odometry" />
        <remap from="/db_pose" to="roomba6_odom/db_pose" />
        <remap from="/db_poses" to="roomba6_odom/db_poses" />
        <remap from="/score" to="roomba6_odom/score" />
        <remap from="/path" to="roomba6_odom/path" />

        <param name="roomba_name" value="6" type="int" />
        <param name="num_s" value="1.0" type="double" />
        <param name="weight_s" value="1.5" type="double" />
        <param name="particlce_num" value="300" type="int" />
        <param name="x_init" value="7.0" type="double" />
        <param name="y_init" value="-1.5" type="double" />
        <param name="yaw_init" value="1.57" type="double" />
        <param name="bench" value="1.0" type="double" />
        <param name="fire" value="1.0" type="double" />
        <param name="big" value="1.0" type="double" />
        <param name="trash" value="0.8" type="double" />
        <param name="odom" value="20" type="double" />
        <param name="alpha_slow_th" value="0.001" type="double" />
        <param name="alpha_fast_th" value="0.1" type="double" />
        <param name="x_cov_init" value="0.8" type="double" />
        <param name="y_cov_init" value="0.8" type="double" />
        <param name="yaw_cov_init" value="0.4" type="double" />
        <param name="x_cov_1" value="0.1" type="double" />
        <param name="y_cov_1" value="0.1" type="double" />
        <param name="yaw_cov_1" value="0.1" type="double" />
        <param name="x_cov_2" value="0.01" type="double" />
        <param name="y_cov_2" value="0.01" type="double" />
        <param name="yaw_cov_2" value="0.02" type="double" />
        <param name="limit" value="100" type="int" />
        <param name="move_noise" value="0.2" type="double" />
        <param name="estimated_weight_th" value="4.0" type="double" />
        <param name="roomba_odom" value="roomba6_odom_odom" type="string" />
        <param name="is_only_odom" value="true" type="bool" />
        </node>
    <!--  -->
    <include file="$(find distance_based_localizer)/launch/evaluation.launch">
        <arg name="roomba" value="1"/>
        <arg name="rec_bag_name" value="roomba1_odom"/>
    </include>
    <include file="$(find distance_based_localizer)/launch/evaluation.launch">
        <arg name="roomba" value="2"/>
        <arg name="rec_bag_name" value="roomba2_odom"/>
    </include>
    <include file="$(find distance_based_localizer)/launch/evaluation.launch">
        <arg name="roomba" value="3"/>
        <arg name="rec_bag_name" value="roomba3_odom"/>
    </include>
    <include file="$(find distance_based_localizer)/launch/evaluation.launch">
        <arg name="roomba" value="4"/>
        <arg name="rec_bag_name" value="roomba4_odom"/>
    </include>
    <include file="$(find distance_based_localizer)/launch/evaluation.launch">
        <arg name="roomba" value="5"/>
        <arg name="rec_bag_name" value="roomba5_odom"/>
    </include>
    <include file="$(find distance_based_localizer)/launch/evaluation.launch">
        <arg name="roomba" value="6"/>
        <arg name="rec_bag_name" value="roomba6_odom"/>
    </include>
</launch>

